### DISCUSSION TITLE
Conclusion: don't install guzzlehttp/guzzle 6.5.x-dev For Upgrading 7 from 6

### DISCUSSION THREAD

**Discussion by choudharyafsaralikhurshid** [kreait/firebase-php#566] [General]
Getting error when laravel upgrading 7 from 6 used this require and require-dev 
"require": {
        "php": ">=7.2.5",
        "bugsnag/bugsnag-laravel": "^2.18",
        "fideloper/proxy": "^4.2",
        "fruitcake/laravel-cors": "^1.0",
        "guzzlehttp/guzzle": "^7.0",
        "intervention/image": "^2.4",
        "jrm2k6/cloudder": "0.6.*",
        "kreait/firebase-php":  "^4.44.0",
        "kreait/gcp-metadata":  "^1.1.0",
        "kreait/firebase-tokens": "^1.10",
        "kreait/laravel-firebase": "^1.5",
        "laravel/framework": "^7.0",
        "laravel/helpers": "^1.2",
        "laravel/passport": "^9.0",
        "laravel/slack-notification-channel": "^2.1",
        "laravel/tinker": "^2.0",
        "laravel/vapor-core": "^2.5",
        "league/flysystem-aws-s3-v3": "~1.0",
        "phpseclib/phpseclib": "^2.0",
        "prettus/l5-repository": "^2.7",
        "prettus/laravel-validation": "^1.3",
        "sofa/eloquence-base": "^7.0",
        "spatie/laravel-medialibrary": "^8.7",
        "tamayo/laravel-scout-elastic": "^5.0"
    },
    "require-dev": {
        "lanin/laravel-api-debugger": "^4.0",
        "appointer/swaggervel": "dev-master",
        "beyondcode/laravel-dump-server": "^1.4",
        "doctrine/dbal": "~2.10",
        "filp/whoops": "^2.0",
        "fzaninotto/faker": "^1.8",
        "infyomlabs/laravel-generator": "^6.0",
        "infyomlabs/swagger-generator": "dev-master",
        "laravel/homestead": "^9.2",
        "mockery/mockery": "^1.4",
        "mpociot/laravel-test-factory-helper": "^2.1",
        "nunomaduro/collision": "^4.1",
        "phpunit/phpunit": "^8.5"
    },

**Comment by choudharyafsaralikhurshid** [comment]
Getting error
Problem 1
    - kreait/gcp-metadata 1.1.0 requires guzzlehttp/guzzle ^6.0 -> satisfiable by guzzlehttp/guzzle[6.5.x-dev].
    - kreait/gcp-metadata 1.1.0 requires guzzlehttp/guzzle ^6.0 -> satisfiable by guzzlehttp/guzzle[6.5.x-dev].
    - Conclusion: don't install guzzlehttp/guzzle 6.5.x-dev
    - Installation request for kreait/gcp-metadata ^1.1.0 -> satisfiable by kreait/gcp-metadata[1.1.0].

**Comment by jeromegamez** [comment]
You can safely remove the `kreait/gcp-metadata` dependency from your root `composer.json`

**Reply by choudharyafsaralikhurshid** [reply]
"kreait/firebase-php":  "^4.44.0",
 "kreait/gcp-metadata":  "^1.1.0",
when removed these getting this error
Problem 1
    - kreait/laravel-firebase 1.5.0 requires kreait/firebase-php ^4.40.1 -> satisfiable by kreait/firebase-php[4.44.0, 4.40.1, 4.41.0, 4.42.0, 4.42.1, 4.43.0, 4.x-dev].
    - kreait/laravel-firebase 1.5.0 requires kreait/firebase-php ^4.40.1 -> satisfiable by kreait/firebase-php[4.44.0, 4.40.1, 4.41.0, 4.42.0, 4.42.1, 4.43.0, 4.x-dev].
    - kreait/firebase-php 4.44.0 requires kreait/gcp-metadata ^1.1.0 -> satisfiable by kreait/gcp-metadata[1.1.0].
    - kreait/firebase-php 4.40.1 requires kreait/gcp-metadata ^1.0.1 -> satisfiable by kreait/gcp-metadata[1.1.0, 1.0.1].
    - kreait/firebase-php 4.41.0 requires kreait/gcp-metadata ^1.0.1 -> satisfiable by kreait/gcp-metadata[1.1.0, 1.0.1].
    - kreait/firebase-php 4.42.0 requires kreait/gcp-metadata ^1.1.0 -> satisfiable by kreait/gcp-metadata[1.1.0].
    - kreait/firebase-php 4.42.1 requires kreait/gcp-metadata ^1.1.0 -> satisfiable by kreait/gcp-metadata[1.1.0].
    - kreait/firebase-php 4.43.0 requires kreait/gcp-metadata ^1.1.0 -> satisfiable by kreait/gcp-metadata[1.1.0].
    - kreait/firebase-php 4.44.0 requires kreait/gcp-metadata ^1.1.0 -> satisfiable by kreait/gcp-metadata[1.1.0].
    - kreait/firebase-php 4.x-dev requires kreait/gcp-metadata ^1.1.0 -> satisfiable by kreait/gcp-metadata[1.1.0].
    - kreait/gcp-metadata 1.1.0 requires guzzlehttp/guzzle ^6.0 -> satisfiable by guzzlehttp/guzzle[6.5.x-dev].
    - kreait/gcp-metadata 1.1.0 requires guzzlehttp/guzzle ^6.0 -> satisfiable by guzzlehttp/guzzle[6.5.x-dev].
    - kreait/gcp-metadata 1.0.1 requires guzzlehttp/guzzle ^6.0 -> satisfiable by guzzlehttp/guzzle[6.5.x-dev].
    - Conclusion: don't install guzzlehttp/guzzle 6.5.x-dev
    - Installation request for kreait/laravel-firebase ^1.5 -> satisfiable by kreait/laravel-firebase[1.5.0].

**Reply by jeromegamez** [reply]
The current version of the Laravel Package is 3.1 - perhaps you would like to go with that one.

https://github.com/kreait/laravel-firebase

**Reply by choudharyafsaralikhurshid** [reply]
![Screenshot from 2021-02-10 18-11-01](https://user-images.githubusercontent.com/26706193/107751965-15731780-6d44-11eb-8834-116bda07a597.png)
Now getting this new error

**Reply by jeromegamez** [reply]
Perhaps I should have made it more clear: Please don't install the 1.5 version of the Laravel package - it's outdated and it will not work.

Remove the following lines from your composer.json

```
"kreait/firebase-php": "^4.44.0",
"kreait/gcp-metadata": "^1.1.0",
"kreait/firebase-tokens": "^1.10",
"kreait/laravel-firebase": "^1.5",
```

and then, do a new `composer require kreait/laravel-firebase` from the terminal. This should install the latest version of the laravel firebase package (3.1.0).

**Reply by choudharyafsaralikhurshid** [reply]
but this package use for last time
"kreait/firebase-tokens": "^1.9"

**Reply by jeromegamez** [reply]
I'm not sure I understand. `kreait/laravel-firebase` requires `kreait/firebase-php`, and `kreait/firebase-php` requires `kreait/firebase-tokens`.

https://github.com/kreait/firebase-php/blob/5.14.1/composer.json#L32
https://github.com/kreait/laravel-firebase/blob/3.1.0/composer.json#L15

The `kreait/gcp-metadata` package is not used anymore

**Reply by choudharyafsaralikhurshid** [reply]
Getting this error when removed above package and run laravel-firebase
Problem 1
    - The requested package guzzlehttp/guzzle (locked at 6.5.5, required as ^7.0) is satisfiable by guzzlehttp/guzzle[6.5.5] but these conflict with your requirements or minimum-stability.
  Problem 2
    - The requested package laravel/framework (locked at v6.18.23, required as ^7.0) is satisfiable by laravel/framework[v6.18.23] but these conflict with your requirements or minimum-stability.
  Problem 3
    - The requested package laravel/passport (locked at v7.5.1, required as ^9.0) is satisfiable by laravel/passport[v7.5.1] but these conflict with your requirements or minimum-stability.
  Problem 4
    - The requested package laravel/tinker (locked at v1.0.10, required as ^2.0) is satisfiable by laravel/tinker[v1.0.10] but these conflict with your requirements or minimum-stability.
  Problem 5
    - The requested package prettus/l5-repository (locked at 2.6.46, required as ^2.7) is satisfiable by prettus/l5-repository[2.6.46] but these conflict with your requirements or minimum-stability.
  Problem 6
    - The requested package prettus/laravel-validation (locked at 1.2.2, required as ^1.3) is satisfiable by prettus/laravel-validation[1.2.2] but these conflict with your requirements or minimum-stability.
  Problem 7
    - The requested package sofa/eloquence-base (locked at v6.0, required as ^7.0) is satisfiable by sofa/eloquence-base[v6.0] but these conflict with your requirements or minimum-stability.
  Problem 8
    - The requested package spatie/laravel-medialibrary (locked at 7.19.3, required as ^8.7) is satisfiable by spatie/laravel-medialibrary[7.19.3] but these conflict with your requirements or minimum-stability.
  Problem 9
    - The requested package lanin/laravel-api-debugger (locked at 3.5.1, required as ^4.0) is satisfiable by lanin/laravel-api-debugger[3.5.1] but these conflict with your requirements or minimum-stability.
  Problem 10
    - The requested package beyondcode/laravel-dump-server (locked at 1.3.0, required as ^1.4) is satisfiable by beyondcode/laravel-dump-server[1.3.0] but these conflict with your requirements or minimum-stability.
  Problem 11
    - The requested package laravel/homestead (locked at v8.6.0, required as ^9.2) is satisfiable by laravel/homestead[v8.6.0] but these conflict with your requirements or minimum-stability.
  Problem 12
    - The requested package mockery/mockery (locked at 1.3.1, required as ^1.4) is satisfiable by mockery/mockery[1.3.1] but these conflict with your requirements or minimum-stability.
  Problem 13
    - The requested package nunomaduro/collision (locked at v2.1.1, required as ^4.1) is satisfiable by nunomaduro/collision[v2.1.1] but these conflict with your requirements or minimum-stability.
  Problem 14
    - The requested package phpunit/phpunit (locked at 7.5.20, required as ^8.5) is satisfiable by phpunit/phpunit[7.5.20] but these conflict with your requirements or minimum-stability.
  Problem 15
    - kreait/gcp-metadata 1.1.0 requires guzzlehttp/guzzle ^6.0 -> satisfiable by guzzlehttp/guzzle[6.5.x-dev].
    - kreait/gcp-metadata 1.1.0 requires guzzlehttp/guzzle ^6.0 -> satisfiable by guzzlehttp/guzzle[6.5.x-dev].
    - Conclusion: don't install guzzlehttp/guzzle 6.5.x-dev
    - Installation request for kreait/gcp-metadata (locked at 1.1.0) -> satisfiable by kreait/gcp-metadata[1.1.0].

**Reply by jeromegamez** [reply]
If you look at the output you'll see that almost all the dependencies are locked at super-old versions. This is not related to this SDK in particular but is an overall problem.

What I would do is delete the `composer.lock` file and `vendor` directory in your project and then do a `composer update` 🤞🏼

**Reply by choudharyafsaralikhurshid** [reply]
before this composer require kreait/laravel-firebase right?

**Reply by jeromegamez** [reply]
I would remove the lines

```
"kreait/firebase-php": "^4.44.0",
"kreait/gcp-metadata": "^1.1.0",
"kreait/firebase-tokens": "^1.10",
"kreait/laravel-firebase": "^1.5",
```

before, and do a `composer require kreait/laravel-firebase` afterwards

**Reply by choudharyafsaralikhurshid** [reply]
yes but deleted the vendor and composer.lock file first
